plugins/occt/module/vtkF3DOCCTReader.cxx

```diff
@@ -285,6 +315,104 @@ class vtkF3DOCCTReader::vtkInternals
   }
 
 #if F3D_PLUGIN_OCCT_XCAF
+  StyleMap CollectInheritedStyles(const TDF_Label& rootLabel, const TopoDS_Shape& rootShape)
+  {
+    StyleMap inheritedStyles;
+
+    if (rootLabel.IsNull())
+    {
+      return inheritedStyles;
+    }
+
+    /* collect styled shapes from the document */
+    StyleMap collectedStyles;
+    XCAFPrs::CollectStyleSettings(rootLabel, TopLoc_Location(), collectedStyles);
+
+    /* iterate styled shapes and assign style if leaf, otherwise collect if parent node */
+    const TopAbs_ShapeEnum leafType = this->Parent->GetReadWire() ? TopAbs_EDGE : TopAbs_FACE;
+    std::vector<std::pair<TopoDS_Shape, XCAFPrs_Style> > parents;
+    for (StyleMap::Iterator iter(collectedStyles); iter.More(); iter.Next())
+    {
+      const TopoDS_Shape& shape = iter.Key();
+      if (shape.ShapeType() == leafType)
+      {
+        inheritedStyles.Add(shape, iter.Value());
+      }
+      else if (shape.ShapeType() < leafType)
+      {
+        parents.emplace_back(std::make_pair(shape, iter.Value()));
+      }
+    }
+
+    /* sort parents from deepest upwards */
+    std::sort(parents.begin(), parents.end(),
+      [](const std::pair<TopoDS_Shape, XCAFPrs_Style>& a,
+        const std::pair<TopoDS_Shape, XCAFPrs_Style>& b)
+      { return a.first.ShapeType() > b.first.ShapeType(); });
+
+    /* pass down each parent style props to descendent edge/face leaves */
+    const auto passDownToLeaves = [&](TopAbs_ShapeEnum type)
+    {
+      for (const auto& parent : parents)
+      {
+        for (TopExp_Explorer iter(parent.first, type); iter.More(); iter.Next())
+        {
+          try
+          {
+            this->PassDownStyleProps(parent.second, inheritedStyles.ChangeFromKey(iter.Current()));
+          }
+          catch (Standard_NoSuchObject&)
+          {
+            inheritedStyles.Add(iter.Current(), parent.second);
+          }
+        }
+      }
+    };
+
+    passDownToLeaves(TopAbs_FACE);
+
+    if (this->Parent->GetReadWire())
+    {
+      passDownToLeaves(TopAbs_EDGE);
+    }
+
+    /* pass down default style (if any) to all leaves */
+    try
+    {
+      const XCAFPrs_Style& defaultStyle = collectedStyles.FindFromKey(rootShape);
+      for (StyleMap::Iterator iter(inheritedStyles); iter.More(); iter.Next())
+      {
+        XCAFPrs_Style style = iter.Value();
+        this->PassDownStyleProps(defaultStyle, style);
+      }
+    }
+    catch (Standard_NoSuchObject&)
+    {
+      /* ignore */
```

Comment:
no log or warn ?

Reply:
no, it's an expected possibility that the root shape doesn't have a style set, I'll update the comment